client/src/main/java/io/pravega/client/stream/StreamConfiguration.java

```diff
@@ -113,10 +117,20 @@ public static final class StreamConfigurationBuilder {
 
         public StreamConfiguration build() {
             Set<String> tagSet = validateTags(this.tags);
+            validateTimestampAggregationTimeout(this.timestampAggregationTimeout);
             Preconditions.checkArgument(this.rolloverSizeBytes >= 0, String.format("Segment rollover size bytes cannot be less than 0, actual is %s", this.rolloverSizeBytes));
             return new StreamConfiguration(this.scalingPolicy, this.retentionPolicy, this.timestampAggregationTimeout, tagSet, this.rolloverSizeBytes);
         }
 
+        private void validateTimestampAggregationTimeout(long timestampAggregationTimeout) {
+            if (timestampAggregationTimeout == 0) {
+                // timestampAggregationTimeout is not set, setting it to default value.
+                this.timestampAggregationTimeout = getDefaultTimestampAggregationTimeout();
+            } else {
+                Preconditions.checkArgument(timestampAggregationTimeout > 0L, "TimestampAggregationTimeout should be greater than 0");
+            }
```

Comment:
Do you think instead of else block, we can use Preconditions.checkArguments  at first line of method by updating condition **Preconditions.checkArgument(timestampAggregationTimeout >= 0L, "TimestampAggregationTimeout should be greater than equal to 0");**?

Reply:
We still need to check if the timeout value is 0 and set it to Deafult value, if true.

Reply:
That's right, but else part can be removed.

Reply:
Done.